src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -381,18 +381,42 @@ object RuleUtils {
       filesAppended: Seq[Path]): LogicalPlan = {
     // Transform the location of LogicalRelation with appended files.
     val planForAppended = originalPlan transformDown {
-      case baseRelation @ LogicalRelation(fsRelation: HadoopFsRelation, baseOutput, _, _) =>
-        // Set the same output schema with the index plan to merge them using BucketUnion.
-        val updatedOutput =
-          baseOutput.filter(attr => indexSchema.fieldNames.contains(attr.name))
+      case baseRelation @ LogicalRelation(
+            fsRelation @ HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+            baseOutput,
+            _,
+            _) =>
         val newLocation =
-          new InMemoryFileIndex(spark, filesAppended, Map(), None)
-        val newRelation =
-          fsRelation.copy(
-            location = newLocation,
-            dataSchema = StructType(indexSchema.filter(baseRelation.schema.contains(_))),
-            options =
-              fsRelation.options + IndexConstants.INDEX_RELATION_IDENTIFIER)(spark)
+          if (location.partitionSchema.isEmpty) {
+            new InMemoryFileIndex(spark, filesAppended, Map(), None)
+          } else {
+            // Set "basePath" so that partitioned columns are also included in the output schema.
+            val basePath = location.partitionSpec.partitionColumns
+              .foldRight(location.partitionSpec.partitions.head.path) { (col, path) =>
```

Comment:
Can you add a comment to the code why `foldRight` is required here? (e.g., traversing back from the child folder, etc.) Btw, is it guaranteed that `partitionColumns` will be in that order?

Reply:
Btw, wouldn't rootPath be same as basePath if hive-partitioning is used (`location.partitionSchema.nonEmpty`)?

Reply:
https://github.com/apache/spark/blob/d6a68e0b67ff7de58073c176dd097070e88ac831/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala#L220

partitionColumns seems be in order, though it cannot be guaranteed clearly.
But I think using "=" in the condition below is also risky..

Reply:
Ok let me just use the number of partition columns as a partition path always includes all partition column names and handle  possible exceptional cases(?) on demand..